src/compile/axis.ts

```diff
@@ -244,13 +244,40 @@ export namespace properties {
       }, labelsSpec);
     }
 
+    let datum = '{{ datum.data }}';
+
     if (contains([NOMINAL, ORDINAL], fieldDef.type) && axis.labelMaxLength) {
       // TODO replace this with Vega's labelMaxLength once it is introduced
-      labelsSpec = extend({
-        text: {
-          template: '{{ datum.data | truncate:' + axis.labelMaxLength + '}}'
-        }
-      }, labelsSpec || {});
+      datum = '{{ datum.data | truncate:' + axis.labelMaxLength + '}}';
+      if (fieldDef.unitPosition == 'suffix') {
```

Comment:
It looks like we have the same logic twice here. Can you move it after the `if`s?